sources/packages/web/src/services/AuthService.ts

```diff
@@ -93,6 +110,20 @@ export class AuthService {
       });
 
       if (this.keycloak.authenticated) {
+        // In case of user closed browser without logout.
+        window.addEventListener("beforeunload", this.logUserClosedBrowser);
```

Comment:
Due to the amount of checks in the session storage, I would recommend centralizing the logic in some services.
Please see the below example with the 3 methods needed for the user triggering login.
It would encapsulate the constants, the logic, the checks, and all `sessionStorage` related operations, and it would improve the code readability.
I would not see this as a blocker but I would strongly recommend to give it a try.

```ts
const USER_LOGIN_TRIGGERED = "userLoginTriggered";
const USER_CLOSED_BROWSER = "userClosedBrowser";
const USER_SESSION_TIMED_OUT = "userSessionTimedOut";

export class AuditService {
  static userLoginTriggered() {
    sessionStorage.setItem(USER_LOGIN_TRIGGERED, "true");
  }

  static wasUserLoginTriggered() {
    return sessionStorage.getItem(USER_LOGIN_TRIGGERED) === "true";
  }

  static resetLoginTriggered() {
    sessionStorage.removeItem(USER_LOGIN_TRIGGERED);
  }
}
```